### PR TITLE
add support for literal svg

### LINKED ISSUE #231: Reading svg text/xml?
As I was working on reproducible magick logo (#230) I imported a couple of SVGs and tried rendering Rstudio's `fontawesome` icons (which renders valid SVG code). 

It could be useful to have a function that accepts SVG text object (like what `fontawesome::fa()` outputs or what you can define with @coolbutuseless `minisvg` [package](https://github.com/coolbutuseless/minisvg)) or an `xml2` object (like when you read an SVG using `xml2::read_xml`) and can render it directly onto canvas. It is actually requiring only small tweak in `image_read_svg()` to start accepting text and xml, but I wonder if it would be confusing when used inside code like this:

```r
my_image %>%
  image_composite(image_read_svg(fontawesome::fa('r-project', "steelblue"), width=200),
                  operator="Atop")
```
Maybe it would be better to have a separate function `image_svg` which accepts only text/xml (along with width/height attributes to allow instant scaling before rasterization)?

### PR DESCRIPTION
**Pull request by dmi3kno**
Closes #231 adding support for literal svg.
I did not add unit test, only an example with `fontawesome`, but this code runs now, i.e. it can be reused for unit test, if you want.
```r
library(magick)
im_path <- "https://upload.wikimedia.org/wikipedia/commons/9/9a/ImageMagick_logo.svg"
im_logo <- readr::read_file(im_path)
r_logo <- fontawesome::fa('r-project', fill = "steelblue")
image_read_svg(im_logo)
image_read_svg(r_logo)
image_read_svg(list(im_logo, r_logo))
```

### CODE REVIEW

**Comment by jeroen** [comment]
What happens if the user has a character vector which is one big svg file like so:

```r
im_path <- "https://upload.wikimedia.org/wikipedia/commons/9/9a/ImageMagick_logo.svg"
svg <- readLines(im_path)
image_read_svg(svg)
```

**Comment by dmi3kno** [comment]
TL;DR: This is not a valid input. What error should we throw?

There's no way to ensure completeness and validity of svg from text vector. It fails now, but with cryptic error (actually due to wrong regex in `is_url()`, I commented on it in an earlier commit. Regex should read "^http?:").

My validator in `is_svg` will fail in every element of the vector except for the last one. You could argue that I could check not only for closing <\svg> tag but also for opening tag. I could do that, but there will be a risk of stripping down important XML schema definition and comments (which is the case here for image produced with Incscape it leaves the comment in line 2).

We could add an opening tag validation to the regex in `is_svg`. We could also detect opening `<?`  and/or closing `?>` schema tag and raise warning "It looks like you are trying to import an svg, but it is broken down to pieces in your character vector. Paste it together and try again." But then again, svg lines can be broken down in arbitrary way, so you can easily have schema definition over multiple lines.

**Comment by dmi3kno** [comment]
After fixing the `is_url` it will fail in the very first element of the character vector with the following error:
```r
image_read_svg(svg)
#>  Error in svg_data(svg) : 
#>  Argument 'svg' must be a file path, url, or raw vector. 
```
Which I think is fully acceptable
